File: articles/communicating-model-uncertainty/fortran/communication_summary.f90
program communication_summary
  implicit none

  integer, parameter :: n = 5
  character(len=32) :: keys(n), layers(n), audiences(n), statuses(n)
  real(8) :: scores(n)
  integer :: i

  keys = (/ 'central_result                 ', 'uncertainty_range              ', 'threshold_risk                 ', 'structural_limit               ', 'use_limit                      ' /)
  layers = (/ 'result                         ', 'uncertainty                    ', 'decision_threshold             ', 'model_limit                    ', 'governance                     ' /)
  audiences = (/ 'decision_maker                ', 'public                        ', 'decision_maker                ', 'technical_reviewer            ', 'future_user                   ' /)
  statuses = (/ 'active                        ', 'review                        ', 'review                        ', 'review                        ', 'review                        ' /)

  print '(A)', 'key layer audience status priority'

  do i = 1, n
    scores(i) = priority(trim(layers(i)), trim(audiences(i)), trim(statuses(i)))
    print '(A,1X,A,1X,A,1X,A,1X,F6.2)', trim(keys(i)), trim(layers(i)), trim(audiences(i)), trim(statuses(i)), scores(i)
  end do

contains

  real(8) function priority(layer, audience, status)
    implicit none
    character(len=*), intent(in) :: layer, audience, status

    if (status == 'active') then
      priority = 1.0d0
    else
      priority = 5.0d0
    end if

    if (layer == 'decision_threshold' .or. layer == 'governance' .or. layer == 'model_limit') then
      priority = priority + 2.0d0
    end if

    if (audience == 'public' .or. audience == 'decision_maker') then
      priority = priority + 1.0d0
    end if
  end function priority

end program communication_summary

File: articles/communicating-model-uncertainty/cpp/communication_summary.cpp
#include <iomanip>
#include <iostream>
#include <string>
#include <vector>

struct CommunicationRecord {
    std::string key;
    std::string layer;
    std::string audience;
    std::string status;
};

double priority(const CommunicationRecord& record) {
    double score = record.status == "active" ? 1.0 : 5.0;
    if (record.layer == "decision_threshold" || record.layer == "governance" || record.layer == "model_limit") {
        score += 2.0;
    }
    if (record.audience == "public" || record.audience == "decision_maker") {
        score += 1.0;
    }
    return score;
}

int main() {
    std::vector<CommunicationRecord> records = {
        {"central_result", "result", "decision_maker", "active"},
        {"uncertainty_range", "uncertainty", "public", "review"},
        {"threshold_risk", "decision_threshold", "decision_maker", "review"},
        {"structural_limit", "model_limit", "technical_reviewer", "review"},
        {"use_limit", "governance", "future_user", "review"}
    };

    std::cout << "key,communication_layer,audience,status,priority\n";
    for (const auto& record : records) {
        std::cout << std::fixed << std::setprecision(2)
                  << record.key << "," << record.layer << "," << record.audience << ","
                  << record.status << "," << priority(record) << "\n";
    }

    return 0;
}
